### DISCUSSION TITLE
Learning more but not sure what!

### DISCUSSION THREAD

**Discussion by happybeing** [OkuBrowser/oku#291] [General]
Hi Emil,

I stumbled on Oku a day after I read a bit about Iroh. Two projects that have a lot of overlap with things I've been dabbling in for a decade and strangely hadn't come across before. I've recently decided to step back from what I was doing and see what appeals but would like to share a bit of that to give some context before asking about Oku.

I've been re-learning software tooling over the past decade of getting ever more convinced we need to build on peer-to-peer to preserve our privacy and autonomy in an increasingly oppressive corporate owned online space.

That's a long journey so I will skip to now where I've spent this year building a demo web browser using Tauri, Svelte and Rust to show how a versioned web (and file system) could be built on top of one p2p project. I have that working save for issues with the platform I chose (Autonomi) which is still buggy but developing fast.

What I was looking for when trying to build my project 'awe' was a Rust based browser that I could tweak, but I couldn't use Servo and didn't find anything else so ended up using Tauri to build my own very crude browser just to demo the concept. `awe` works as well as it can and can publish websites to Autonomi, and when Autonomi behaves anyone can view every published version of a website proving the concept. Autonomi is a p2p storage network with an emphasis on privacy and data preservation which makes it much more suitable for something like the web (always on forever, no surveillance, anonymous access for anyone from anywhere, no gatekeepers who decide what can be published or viewed etc). I didn't find anything else close to as good as Autonomi on those issues all the time I've been doing this.

I don't know how I missed Oku when looking for a way to build `awe` but I might still not have chosen to try using it because I wanted to demo across as many platforms as possible, including mobile. However, I have learned that doing this is more difficult than I hoped for various reasons, so for now `awe` only works on Linux and MacOS, although it does build for Windows and Android thanks to Tauri v2's support.

Working alone is my thing really, but I still wanted to reach out and see if you would like a bit of discussion about Oku, maybe Iroh and perhaps the things I want to demo in `awe`: a versioned, privacy respecting perpetual web with no gatekeepers to publishing or viewing.  How important are those kind of issues to you? I'd like to understand more about what you think is important, not just functionality but the kinds of problems you see right now with the online space and things you would like to see to improve it.

I was very please to see you also use AGPL, not least because when I looked at Iroh I wasn't impressed to see it using Apache and that n0 might just be another commercial play that would turn everything closed source when it suits them. So I've asked them some q's on Mastodon but waiting for a response.

I tend to write too much! So I will stop and here's a [link to awe](https://github.com/happybeing/awe) in case you want to get a feel for my work (lots of related demos there too such as a PoC for a peer-to-peer version of github: p2pgit).

All the best,

happybeing

**Comment by emmyoh** [comment]
Thanks for reaching out! Have a lot of thoughts and will reply with more when I get the chance/time. Just want you to know that I've read this :)
